sdk/go/pulumi/types_test.go

```diff
@@ -301,3 +309,86 @@ func TestToOutputInputAny(t *testing.T) {
 		A: map[string]interface{}{"world": true},
 	}, v)
 }
+
+// Test that SecretT sets appropriate internal state and that IsSecret appropriately reads it.
+func TestSecrets(t *testing.T) {
+	s := SecretT(String("foo"))
+	// assert that secret is immediately secret
+	assert.True(t, s.IsSecret())
+
+	errChan := make(chan error)
+	resultChan := make(chan string)
+	secretChan := make(chan bool)
```

Comment:
You should be able to use `await` in this test.

Reply:
True, but going to leave these as-is.